domain/schema_validator.go

```diff
@@ -234,8 +234,9 @@ func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedPhysicalTa
 
 	// Schema changed, result decided by whether related tables change.
 	if schemaVer < s.latestSchemaVer {
-		// The DDL relatedPhysicalTableIDs is empty.
-		if len(relatedPhysicalTableIDs) == 0 {
+		// When a transaction executes a DDL, relatedPhysicalTableIDs is nil.
+		// When a transaction only contains DML on temporary tables, relatedPhysicalTableIDs is [].
+		if relatedPhysicalTableIDs == nil {
```

Comment:
Why not just use `len(relatedPhysicalTableIDs) == 0`? It looks more intuitive.
It's a burden for the reader to distinguish `nil` and `[]` and search the corresponding context to understand the code.

Reply:
The reason is in the comment. 
The code block leads to failure, but I don't want it to fail where there's nothing to commit.
Any better ideas?
I also tried to remove SchemaChecker when there are no normal tables to commit, but some logic needs a SchemaChecker.

Reply:
This time I understand their difference.
`relatedPhysicalTableIDs == nil` => deny
`relatedPhysicalTableIDs == []` => accept